src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -24,6 +24,11 @@ public WritableFileChooser(File currentDirectory) {
 	@Override
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
+		
+		if (selectedFile.length() > selectedFile.getUsableSpace()) {
```

Comment:
That makes sense. I initially imagined a check to verify that at least 5-10MB (possible user set limit?) exists on a hard disk before proceeding, but I figured that method would be sloppy. I guess for now I could do more research into it unless someone else has an idea.